packages/networking/test/unit/agent_spec.ts

```diff
@@ -0,0 +1,413 @@
+import * as http from 'http'
+import * as https from 'https'
+import * as net from 'net'
+import * as tls from 'tls'
+import * as url from 'url'
+import * as chai from 'chai'
+import DebuggingProxy = require('debugging-proxy')
+import * as Promise from 'bluebird'
+import * as request from 'request-promise'
+import * as sinon from 'sinon'
+import * as sinonChai from 'sinon-chai'
+import { agent } from '../..'
+import * as Io from '@packages/socket'
+import { Servers, AsyncServer } from '../support/servers'
+
+const expect = chai.expect
+chai.use(sinonChai)
+
+const PROXY_PORT = 31000
+const HTTP_PORT = 31080
+const HTTPS_PORT = 31443
+
+describe('lib/agent', function() {
```

Comment:
These tests where we are starting servers and making requests are really `integration` tests - whereas the stateless / pure tests on the exported functions are `unit` tests.

Not saying to break it out, but it may help separate these things.